src/SimpleJson.ts

```diff
@@ -0,0 +1,325 @@
+export class SimpleJson {
+	public static TextToDictionary(text: string) {
+		return new SimpleJson.Reader(text).ToDictionary();
+	}
+
+	public static TextToArray(text: string) {
+		return new SimpleJson.Reader(text).ToArray();
+	}
+}
+
+export namespace SimpleJson {
+	export class Reader {
+		constructor(text: string) {
+			this._rootObject = JSON.parse(text);
+		}
+
+		public ToDictionary() {
+			return this._rootObject as Record<string, any>;
+		}
+
+		public ToArray() {
+			return this._rootObject as any[];
+		}
+
+		private _rootObject: any[] | Record<string, any>;
```

Comment:
It's not indeed, I didn't want to touch what worked just yet. I'll update this at the very end!

There's less because we're using the built-in JSON parser, while C# uses a custom one.